### ISSUE TITLE
tuplets are broken when stream contains grace notes

### ISSUE THREAD

**Issue by gesellkammer** [cuthbertLab/music21#344]
The problem is in makeTupletBrackets, which does not skip grace notes (notes with 0 duration) when calculating brackets.

This fixes the issue (see durationList = ...)

```python

def makeTupletBrackets(s):
    durationList = []
    returnObj = copy.deepcopy(s)
    # only want to look at notes
    durationList = [n.duration for n in returnObj.notesAndRests if n.duration.quarterLength > 0]
    
    tupletMap = [] # a list of (tuplet obj / Duration) pairs
    for dur in durationList: # all Duration objects
        tupletList = dur.tuplets
        if tupletList in [(), None]: # no tuplets, length is zero
            tupletMap.append([None, dur])
        elif len(tupletList) > 1:
            environLocal.warn('got multi-tuplet duration; cannot yet handle this. %s' %
                              repr(tupletList))
        elif len(tupletList) == 1:
            tupletMap.append([tupletList[0], dur])
            if tupletList[0] != dur.tuplets[0]:
                raise Exception('cannot access Tuplets object from within DurationTuple.')
        else:
            raise Exception('cannot handle these tuplets: %s' % tupletList)

    # have a list of tuplet, Duration pairs
    completionCount = 0      # qLen currently filled
    completionTarget = None  # qLen necessary to fill tuplet
    for i in range(len(tupletMap)):
        tupletObj, dur = tupletMap[i]
        tupletPrevious = tupletMap[i-1][0] if i > 0 else None
        tupletNext = tupletMap[i+1][0] if i < len(tupletMap)-1 else None
        if tupletObj is not None:
            completionCount = opFrac(completionCount + dur.quarterLength)
            # if previous tuplet is None, always start
            # always reset completion target
            if tupletPrevious is None or completionTarget is None:
                if tupletNext is None: # single tuplet w/o tuplets either side
                    tupletObj.type = 'startStop'
                    tupletObj.bracket = False
                    completionCount = 0 # reset
                else:
                    tupletObj.type = 'start'
                    completionTarget = tupletObj.totalTupletLength()        
            # if tuplet next is None, always stop
            # if both previous and next are None, just keep a start

            # this, below, is optional:
            # if next normal type is not the same as this one, also stop
            elif (tupletNext is None or completionCount >= completionTarget):
                tupletObj.type = 'stop' # should be impossible once frozen...
                completionTarget = None # reset
                completionCount = 0     # reset                
            elif tupletPrevious is not None and tupletNext is not None:
                # do not need to change tuplet type; should be None
                pass
                
    return returnObj
```

**Comment by jacobtylerwalls** [comment]
fixed in f23fa76de0

**Comment by mscuthbert** [comment]
Thanks Jacob!
